# Project: Wrangle WeRateDogs data from Twitter

Table of Contents
1. Introduction
2. Data Wrangling
3. Storing and Acting on Wrangling Data

1.Introduction
This project is about wrangling data for the Wrangle and Analyze Project
The document is divided in 3 sections, each for one step of the dta wrangling process that we saw in the lessons

Context
Our goal: wrangle WeRateDogs Twitter data to create interesting and trustworthy analyses and visualizations. The Twitter archive is great, but it only contains very basic tweet information. Additional gathering, then assessing and cleaning is required for "Wow!"-worthy analyses and visualizations.

Project Details
Our tasks in this project are as follows:

Data wrangling, which consists of:
Gathering data (downloadable file in the Resources tab in the left most panel of your classroom and linked in step 1 below).
Assessing data
Cleaning data
Storing, analyzing, and visualizing your wrangled data
Reporting on 1) your data wrangling efforts and 2) your data analyses and visualizations


2. Data Wrangling

Gather
First we need to gather the data from at least 3 different sources. Some data were wprovides and other were obtained

In [14]:
#first load all the libraries needed
import pandas as pd
import numpy as np
import requests
import json

In [1]:
import tweepy

consumer_key = 'XXX'
consumer_secret = 'XXX'
access_token = 'XXX'
access_secret = 'XXX'

auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_secret)

api = tweepy.API(auth)

In [15]:
#Now load the twitter_archive_enhanced.csv file into the dataframe
tweet_data = pd.read_csv("twitter_archive_enhanced.csv")

In [16]:
# Now, we etrieve image-predictions.tsv file
prediction_url = 'https://d17h27t6h515a5.cloudfront.net/topher/2017/August/599fd2ad_image-predictions/image-predictions.tsv'
with open('image-predictions.tsv', 'wb') as prediction_file:
    prediction_gather = requests.get(prediction_url)
    prediction_file.write(prediction_gather.content)
    
#Now we retrieve the image_predictions.tsv file into its dataframe
prediction_data = pd.read_csv("image_predictions.tsv",sep='\t')

Assess

Assess
Assessing our data is the second step in data wrangling. When assessing, we are like a detective at work, inspecting our dataset for two things: data quality issues (i.e. content issues) and lack of tidiness (i.e. structural issues).

Quality issues are issues with content, like inaccurate or duplicate data. Tidiness issues are structural issues, specifically: each variable must be a column, each observation must be a row, and each type of observational unit must be a table.

We will use two steps: Visual Assessment and Programmatic Assessment

1. Dataframe contents for visual assessment

In [17]:
# Display the twitter data dataframe
tweet_data

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
5     891087950875897856                    NaN                  NaN   
6     890971913173991426                    NaN                  NaN   
7     890729181411237888                    NaN                  NaN   
8     890609185150312448                    NaN                  NaN   
9     890240255349198849                    NaN                  NaN   
10    890006608113172480                    NaN                  NaN   
11    889880896479866881                    NaN                  NaN   
12    889665388333682689                    NaN                  NaN   
13    889638837579907072                    NaN                  NaN   
14    889531135344209921                    NaN                  NaN   
15    889278841981685760                    NaN                  NaN   
16    888917238123831296                    NaN                  NaN   
17    888804989199671297                    NaN                  NaN   
18    888554962724278272                    NaN                  NaN   
19    888202515573088257                    NaN                  NaN   
20    888078434458587136                    NaN                  NaN   
21    887705289381826560                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
23    887473957103951883                    NaN                  NaN   
24    887343217045368832                    NaN                  NaN   
25    887101392804085760                    NaN                  NaN   
26    886983233522544640                    NaN                  NaN   
27    886736880519319552                    NaN                  NaN   
28    886680336477933568                    NaN                  NaN   
29    886366144734445568                    NaN                  NaN   
...                  ...                    ...                  ...   
2326  666411507551481857                    NaN                  NaN   
2327  666407126856765440                    NaN                  NaN   
2328  666396247373291520                    NaN                  NaN   
2329  666373753744588802                    NaN                  NaN   
2330  666362758909284353                    NaN                  NaN   
2331  666353288456101888                    NaN                  NaN   
2332  666345417576210432                    NaN                  NaN   
2333  666337882303524864                    NaN                  NaN   
2334  666293911632134144                    NaN                  NaN   
2335  666287406224695296                    NaN                  NaN   
2336  666273097616637952                    NaN                  NaN   
2337  666268910803644416                    NaN                  NaN   
2338  666104133288665088                    NaN                  NaN   
2339  666102155909144576                    NaN                  NaN   
2340  666099513787052032                    NaN                  NaN   
2341  666094000022159362                    NaN                  NaN   
2342  666082916733198337                    NaN                  NaN   
2343  666073100786774016                    NaN                  NaN   
2344  666071193221509120                    NaN                  NaN   
2345  666063827256086533                    NaN                  NaN   
2346  666058600524156928                    NaN                  NaN   
2347  666057090499244032                    NaN                  NaN   
2348  666055525042405380                    NaN                  NaN   
2349  666051853826850816                    NaN         

Visual
Quality Issues
1. Name column, have "None" as name. This can be confusing
2. There is missing data in "name" columns and is written as the string "None"

Tidity Issues
1. The columns "doggo", "floofer", "pupper" and "puppo" represent the same variable "type"

In [18]:
# Display the prediction data dataframe
prediction_data

tweet_id                                            jpg_url  \
0     666020888022790149    https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1     666029285002620928    https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2     666033412701032449    https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   
3     666044226329800704    https://pbs.twimg.com/media/CT5Dr8HUEAA-lEu.jpg   
4     666049248165822465    https://pbs.twimg.com/media/CT5IQmsXIAAKY4A.jpg   
5     666050758794694657    https://pbs.twimg.com/media/CT5Jof1WUAEuVxN.jpg   
6     666051853826850816    https://pbs.twimg.com/media/CT5KoJ1WoAAJash.jpg   
7     666055525042405380    https://pbs.twimg.com/media/CT5N9tpXIAAifs1.jpg   
8     666057090499244032    https://pbs.twimg.com/media/CT5PY90WoAAQGLo.jpg   
9     666058600524156928    https://pbs.twimg.com/media/CT5Qw94XAAA_2dP.jpg   
10    666063827256086533    https://pbs.twimg.com/media/CT5Vg_wXIAAXfnj.jpg   
11    666071193221509120    https://pbs.twimg.com/media/CT5cN_3WEAAlOoZ.jpg   
12    666073100786774016    https://pbs.twimg.com/media/CT5d9DZXAAALcwe.jpg   
13    666082916733198337    https://pbs.twimg.com/media/CT5m4VGWEAAtKc8.jpg   
14    666094000022159362    https://pbs.twimg.com/media/CT5w9gUW4AAsBNN.jpg   
15    666099513787052032    https://pbs.twimg.com/media/CT51-JJUEAA6hV8.jpg   
16    666102155909144576    https://pbs.twimg.com/media/CT54YGiWUAEZnoK.jpg   
17    666104133288665088    https://pbs.twimg.com/media/CT56LSZWoAAlJj2.jpg   
18    666268910803644416    https://pbs.twimg.com/media/CT8QCd1WEAADXws.jpg   
19    666273097616637952    https://pbs.twimg.com/media/CT8T1mtUwAA3aqm.jpg   
20    666287406224695296    https://pbs.twimg.com/media/CT8g3BpUEAAuFjg.jpg   
21    666293911632134144    https://pbs.twimg.com/media/CT8mx7KW4AEQu8N.jpg   
22    666337882303524864    https://pbs.twimg.com/media/CT9OwFIWEAMuRje.jpg   
23    666345417576210432    https://pbs.twimg.com/media/CT9Vn7PWoAA_ZCM.jpg   
24    666353288456101888    https://pbs.twimg.com/media/CT9cx0tUEAAhNN_.jpg   
25    666362758909284353    https://pbs.twimg.com/media/CT9lXGsUcAAyUFt.jpg   
26    666373753744588802    https://pbs.twimg.com/media/CT9vZEYWUAAlZ05.jpg   
27    666396247373291520    https://pbs.twimg.com/media/CT-D2ZHWIAA3gK1.jpg   
28    666407126856765440    https://pbs.twimg.com/media/CT-NvwmW4AAugGZ.jpg   
29    666411507551481857    https://pbs.twimg.com/media/CT-RugiWIAELEaq.jpg   
...                  ...                                                ...   
2045  886366144734445568    https://pbs.twimg.com/media/DE0BTnQUwAApKEH.jpg   
2046  886680336477933568    https://pbs.twimg.com/media/DE4fEDzWAAAyHMM.jpg   
2047  886736880519319552    https://pbs.twimg.com/media/DE5Se8FXcAAJFx4.jpg   
2048  886983233522544640    https://pbs.twimg.com/media/DE8yicJW0AAAvBJ.jpg   
2049  887101392804085760    https://pbs.twimg.com/media/DE-eAq6UwAA-jaE.jpg   
2050  887343217045368832  https://pbs.twimg.com/ext_tw_video_thumb/88734...   
2051  887473957103951883    https://pbs.twimg.com/media/DFDw2tyUQAAAFke.jpg   
2052  887517139158093824  https://pbs.twimg.com/ext_tw_video_thumb/88751...   
2053  887705289381826560    https://pbs.twimg.com/media/DFHDQBbXgAEqY7t.jpg   
2054  888078434458587136    https://pbs.twimg.com/media/DFMWn56WsAAkA7B.jpg   
2055  888202515573088257    https://pbs.twimg.com/media/DFDw2tyUQAAAFke.jpg   
2056  888554962724278272    https://pbs.twimg.com/media/DFTH_O-UQAACu20.jpg   
2057  888804989199671297    https://pbs.twimg.com/media/DFWra-3VYAA2piG.jpg   
2058  888917238123831296    https://pbs.twimg.com/media/DFYRgsOUQAARGhO.jpg   
2059  889278841981685760  https://pbs.twimg.com/ext_tw_video_thumb/88927...   
2060  889531135344209921    https://pbs.twimg.com/media/DFg_2PVW0AEHN3p.jpg   
2061  889638837579907072    https://pbs.twimg.com/media/DFihzFfXsAYGDPR.jpg   
2062  889665388333682689    https://pbs.twimg.com/media/DFi579UWsAAatzw.jpg   
2063  889880896479866881    https://pbs.twimg.com/media/DFl99B1WsA

Visual Quality Issues

3. In p1, p2, and p3 the type of dogs uses underscores that are diffucult to read

Tidity Issues

2. Correct predictions of dogs should be combined with tweet_data

2. Dataframe contents for programmatic assessment

In [19]:
#We display the structure of the dataframe tweet_data
tweet_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 17 columns):
tweet_id                      2356 non-null int64
in_reply_to_status_id         78 non-null float64
in_reply_to_user_id           78 non-null float64
timestamp                     2356 non-null object
source                        2356 non-null object
text                          2356 non-null object
retweeted_status_id           181 non-null float64
retweeted_status_user_id      181 non-null float64
retweeted_status_timestamp    181 non-null object
expanded_urls                 2297 non-null object
rating_numerator              2356 non-null int64
rating_denominator            2356 non-null int64
name                          2356 non-null object
doggo                         2356 non-null object
floofer                       2356 non-null object
pupper                        2356 non-null object
puppo                         2356 non-null object
dtypes: float64(4), int64(3), ob

Quality Issues
4. Missing "expanded URLS", there are 2297 instead of 2356
5. retweeted_status_id need to be "int64"
6. timestamp is better to be "date_time" objet type
7. there are retweet data, it shouldn´t

In [20]:
#look for more issues with sample rows
tweet_data.sample(15)

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
1566  687841446767013888                    NaN                  NaN   
698   786363235746385920                    NaN                  NaN   
395   825535076884762624                    NaN                  NaN   
1967  673342308415348736                    NaN                  NaN   
861   763103485927849985                    NaN                  NaN   
1603  685943807276412928                    NaN                  NaN   
145   863553081350529029                    NaN                  NaN   
1022  746542875601690625                    NaN                  NaN   
1320  706346369204748288                    NaN                  NaN   
93    874057562936811520                    NaN                  NaN   
1655  683391852557561860                    NaN                  NaN   
621   796116448414461957                    NaN                  NaN   
1024  746507379341139972                    NaN                  NaN   
540   806542213899489280                    NaN                  NaN   
1255  710609963652087808                    NaN                  NaN   

                      timestamp  \
1566  2016-01-15 03:39:15 +0000   
698   2016-10-13 00:29:39 +0000   
395   2017-01-29 02:44:34 +0000   
1967  2015-12-06 03:24:51 +0000   
861   2016-08-09 20:03:43 +0000   
1603  2016-01-09 21:58:42 +0000   
145   2017-05-14 00:34:33 +0000   
1022  2016-06-25 03:17:46 +0000   
1320  2016-03-06 05:11:12 +0000   
93    2017-06-12 00:15:36 +0000   
1655  2016-01-02 20:58:09 +0000   
621   2016-11-08 22:25:27 +0000   
1024  2016-06-25 00:56:43 +0000   
540   2016-12-07 16:53:43 +0000   
1255  2016-03-17 23:33:12 +0000   

                                                 source  \
1566  <a href="http://vine.co" rel="nofollow">Vine -...   
698   <a href="http://twitter.com/download/iphone" r...   
395   <a href="http://twitter.com/download/iphone" r...   
1967  <a href="http://twitter.com/download/iphone" r...   
861   <a href="http://twitter.com/download/iphone" r...   
1603  <a href="http://twitter.com/download/iphone" r...   
145   <a href="http://twitter.com/download/iphone" r...   
1022  <a href="http://vine.co" rel="nofollow">Vine -...   
1320  <a href="http://twitter.com/download/iphone" r...   
93    <a href="http://twitter.com/download/iphone" r...   
1655  <a href="http://twitter.com/download/iphone" r...   
621   <a href="http://twitter.com/download/iphone" r...   
1024  <a href="http://twitter.com/download/iphone" r...   
540   <a href="http://twitter.com/download/iphone" r...   
1255  <a href="http://vine.co" rel="nofollow">Vine -...   

                                                   text  retweeted_status_id  \
1566  13/10 I can't stop watching this (vid by @k8ly...                  NaN   
698   This is Rizzo. He has many talents. A true ren...                  NaN   
395   Here's a very loving and accepting puppo. Appe...                  NaN   
1967  This is Django. He's a skilled assassin pupper...                  NaN   
861   This is Belle. She's a Butterflop Hufflepoof. ...                  NaN   
1603  This is the newly formed pupper a capella grou...                  NaN   
145   This is Neptune. He's a backpup vocalist for t...                  NaN   
1022  Here's a golden floofer helping with the groce...                  NaN   
1320  This is Koda. She's a Beneboom Cumberwiggle. 1...                  NaN   
93    I can't believe this keeps happening. This, is...                  NaN   
1655  Say hello to Jack (pronounced "Kevin"). He's a...                  NaN   
621   I didn't believe it at first but now I can see...                  NaN   
1024  This is Shiloh. She did not pass the soft mout...                  NaN   
540   This is Waffles. He's concerned that the dandr...                  NaN   
1255  I've watched this a million times and you prob...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
1566           

In [21]:
# Look for quality issues in the range of data
tweet_data[['rating_numerator', 'rating_denominator']].describe()

rating_numerator  rating_denominator
count       2356.000000         2356.000000
mean          13.126486           10.455433
std           45.876648            6.745237
min            0.000000            0.000000
25%           10.000000           10.000000
50%           11.000000           10.000000
75%           12.000000           10.000000
max         1776.000000          170.000000

Quality Issues
8. There is a minimum value of 0 in the rating_denominator column
9. The maximun value is higher for both numerator and denominator

In [22]:
#Get the value of the numerator column
tweet_data.rating_numerator.value_counts()

12      558
11      464
10      461
13      351
9       158
8       102
7        55
14       54
5        37
6        32
3        19
4        17
1         9
2         9
420       2
0         2
15        2
75        2
80        1
20        1
24        1
26        1
44        1
50        1
60        1
165       1
84        1
88        1
144       1
182       1
143       1
666       1
960       1
1776      1
17        1
27        1
45        1
99        1
121       1
204       1
Name: rating_numerator, dtype: int64

In [23]:
#Dig more for the denominator equals zero
tweet_data.rating_denominator.value_counts()

10     2333
11        3
50        3
80        2
20        2
2         1
16        1
40        1
70        1
15        1
90        1
110       1
120       1
130       1
150       1
170       1
7         1
0         1
Name: rating_denominator, dtype: int64

There is only 1 case that the denominator is "0"

In [24]:
#Looking for duplicated tweet_id
tweet_data.tweet_id.value_counts()

749075273010798592    1
741099773336379392    1
798644042770751489    1
825120256414846976    1
769212283578875904    1
700462010979500032    1
780858289093574656    1
699775878809702401    1
880095782870896641    1
760521673607086080    1
776477788987613185    1
691820333922455552    1
715696743237730304    1
714606013974974464    1
760539183865880579    1
813157409116065792    1
676430933382295552    1
743510151680958465    1
837012587749474308    1
833722901757046785    1
818259473185828864    1
670704688707301377    1
667160273090932737    1
674394782723014656    1
672082170312290304    1
670093938074779648    1
759923798737051648    1
809920764300447744    1
805487436403003392    1
838085839343206401    1
                     ..
763956972077010945    1
870308999962521604    1
720775346191278080    1
785927819176054784    1
783347506784731136    1
775733305207554048    1
834209720923721728    1
825026590719483904    1
758405701903519748    1
668986018524233728    1
6909388994772213

No duplicate tweet_id

In [25]:
# Display the structure of prediction data dataframe
prediction_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2075 entries, 0 to 2074
Data columns (total 12 columns):
tweet_id    2075 non-null int64
jpg_url     2075 non-null object
img_num     2075 non-null int64
p1          2075 non-null object
p1_conf     2075 non-null float64
p1_dog      2075 non-null bool
p2          2075 non-null object
p2_conf     2075 non-null float64
p2_dog      2075 non-null bool
p3          2075 non-null object
p3_conf     2075 non-null float64
p3_dog      2075 non-null bool
dtypes: bool(3), float64(3), int64(2), object(4)
memory usage: 152.1+ KB


Quality Issues
10. There are 2075 entries and in the tweet_data dataframe there are 2376 rows, so there is missing data

In [26]:
#look for duplicate data in prediction_data dataframe
prediction_data.tweet_id.value_counts()

685532292383666176    1
826598365270007810    1
692158366030913536    1
714606013974974464    1
715696743237730304    1
776477788987613185    1
772114945936949249    1
699775878809702401    1
780858289093574656    1
700462010979500032    1
732726085725589504    1
738883359779196928    1
798644042770751489    1
743510151680958465    1
837012587749474308    1
833722901757046785    1
668620235289837568    1
842765311967449089    1
685315239903100929    1
673686845050527744    1
680473011644985345    1
666051853826850816    1
675853064436391936    1
693231807727280129    1
705475953783398401    1
829449946868879360    1
759923798737051648    1
667160273090932737    1
680934982542561280    1
743545585370791937    1
                     ..
794926597468000259    1
776113305656188928    1
825026590719483904    1
834209720923721728    1
775733305207554048    1
669564461267722241    1
879492040517615616    1
720775346191278080    1
666362758909284353    1
750506206503038976    1
6931556864910008

No duplicate data

In [27]:
# Compute statitics on the p*_conf columns in prediction_data
prediction_data[['img_num', 'p1_conf', 'p2_conf', 'p3_conf']].describe()

img_num      p1_conf       p2_conf       p3_conf
count  2075.000000  2075.000000  2.075000e+03  2.075000e+03
mean      1.203855     0.594548  1.345886e-01  6.032417e-02
std       0.561875     0.271174  1.006657e-01  5.090593e-02
min       1.000000     0.044333  1.011300e-08  1.740170e-10
25%       1.000000     0.364412  5.388625e-02  1.622240e-02
50%       1.000000     0.588230  1.181810e-01  4.944380e-02
75%       1.000000     0.843855  1.955655e-01  9.180755e-02
max       4.000000     1.000000  4.880140e-01  2.734190e-01

No quality Issues, no negative numbers, , range betwwen [0,1]

Summary of Quality and Tidiness Issues

Quality

tweet_data dataframe
1. Name column, have "None" as name. This can be confusing
2. There is missing data in "name" columns and is written as the string "None"
4. Missing "expanded URLS", there are 2297 instead of 2356
5. retweeted_status_id need to be "int64"
6. timestamp is better to be "date_time" objet type
7. there are retweet data, it shouldn´t
8. There is a minimum value of 0 in the rating_denominator column
9. The maximun value is higher for both numerator and denominator

prediction_data dataframe
3. In p1, p2, and p3 the type of dogs uses underscores that are diffucult to read
10. There are 2075 entries and in the tweet_data dataframe there are 2376 rows, so there is missing data

Tidiness

tweet_data dataframe
1. The columns "doggo", "floofer", "pupper" and "puppo" represent the same variable "type"

prediction_data dataframe
2. Correct predictions of dogs should be combined with tweet_data

Clean
Cleaning our data is the third step in data wrangling. It is where we fix the quality and tidiness issues that we identified in the assess step

We will use the data cleaning process: defining, coding, and testing
•	We will address the missing data first
•	We will tackle the tidiness issues next
•	And finally, we will clean up the quality issues

The very first thing to do before any cleaning occurs is to make a copy of each piece of data. All of the cleaning operations will be conducted on this copy so we can still view the original dirty and/or messy dataset later

In [28]:
#create a copy of each dataframe
tweet_data_clean = tweet_data.copy()
prediction_data_clean = prediction_data.copy()

FALTA

Quality

6. timestamp is better to be "date_time" objet type

Define
Convert the timestamp column into datetime

Code

In [29]:
#Changetimestamp column to datetime type
ts_col_names = [t_timestamp for t_timestamp in tweet_data_clean.columns.tolist() if 'timestamp' in t_timestamp]
for ts_col_name in ts_col_names:
    tweet_data_clean[ts_col_name] = pd.to_datetime(tweet_data_clean[ts_col_name])

Test

In [30]:
tweet_data_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 17 columns):
tweet_id                      2356 non-null int64
in_reply_to_status_id         78 non-null float64
in_reply_to_user_id           78 non-null float64
timestamp                     2356 non-null datetime64[ns]
source                        2356 non-null object
text                          2356 non-null object
retweeted_status_id           181 non-null float64
retweeted_status_user_id      181 non-null float64
retweeted_status_timestamp    181 non-null datetime64[ns]
expanded_urls                 2297 non-null object
rating_numerator              2356 non-null int64
rating_denominator            2356 non-null int64
name                          2356 non-null object
doggo                         2356 non-null object
floofer                       2356 non-null object
pupper                        2356 non-null object
puppo                         2356 non-null object
dtypes: datetime

7. there are retweet data, it shouldn´t

Define
Remove all rows where retweeted_status_id. Afterwards, reset the index.

Code

In [31]:
retweeted_cols = ['retweeted_status_id', 'retweeted_status_user_id', 'retweeted_status_timestamp']
retweeted_mask = tweet_data_clean.retweeted_status_id.notnull()

tweet_data_clean = tweet_data_clean[~retweeted_mask]
tweet_data_retweet_test = tweet_data_clean.copy() # For testing
tweet_data_clean.drop(columns=retweeted_cols, inplace=True)
tweet_data_clean.reset_index(drop=True, inplace=True)

Test

In [32]:
# chrck if we deleted all of the retweets
tweet_data_retweet_test.retweeted_status_id.notnull().sum()

0

1. Name column, have "None" as name. This can be confusing

Define
We will replace the string "None" with "NaN"

Code

In [33]:
tweet_data_clean.name = tweet_data_clean.name.replace('None', np.nan)

test

In [34]:
tweet_data_clean.name.loc[tweet_data_clean.name == 'None'].sum()

False

3. Storing and Acting on Wrangling Data

Storing

In [35]:
# save our dataframe in the file specified
tweet_data_clean.to_csv("twitter_archive_master.csv")

In [36]:
# save our dataframe in the file specified
prediction_data_clean.to_csv("twitter_archive_secondary.csv")

Work in progress

Analysis and Insights
3 insights and 1 labeled visualization

In [42]:
#Most common scores on the tweets
tweet_data_clean['score'] = tweet_data_clean['rating_numerator'].astype(str) + "/" + \
                                  tweet_data_clean['rating_denominator'].astype(str)
tweet_data_clean.score.value_counts()

12/10      500
10/10      442
11/10      425
13/10      307
9/10       155
8/10        98
7/10        53
14/10       43
5/10        36
6/10        32
3/10        19
4/10        15
2/10         9
1/10         7
420/10       2
4/20         2
0/10         2
44/40        1
27/10        1
121/110      1
9/11         1
11/15        1
7/11         1
165/150      1
26/10        1
666/10       1
75/10        1
99/90        1
50/50        1
1776/10      1
20/16        1
143/130      1
80/80        1
84/70        1
15/10        1
17/10        1
144/120      1
45/50        1
204/170      1
88/80        1
24/7         1
960/0        1
60/50        1
182/10       1
1/2          1
Name: score, dtype: int64

Insight:
The most common scores are around 10/10 +1 or -1